params.go

```diff
@@ -30,6 +30,58 @@ type Params struct {
 	Data interface{} `json:"-"`
 }
 
+// CommonParams should be used where there are no operation specific Params defined
+type CommonParams struct {
```

Comment:
I'm fuzzy on what this CommonParams vs Params means. CommonParams is just a Params, but then we make a Params pointer when we call toParams()... but CommonParams is already type Params?